NOTES.md

```diff
@@ -0,0 +1,235 @@
+# Librix server API Guide _18/05/2021_
+
+Running locally is set to port 5000.
+
+## Table of Contents
+
+- [Errors](#errors)
+- [Users](#users)
+- [Books](#books)
+- [Matches](#matches)
+
+## Errors
+
+If there is an error in any of the functions we return =>
+
+```
+{
+  "error": {
+    "message": "The error message"
+  }
+}
+```
+
+## Users
+
+All user calls start with /users
+
+### /
+
+**get** => response is an array with the users as objects, this will be deprecated in production of the app
+**post** => expects the following minimal request
+
+```
+{
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "password": "0123abc!@£",
+  "address": {
+	  "city": "Berlin"
+  }
+}
+```
+
+Will recieve the response =>
+
+```
+{
+  "avatar": "/statics/01.png",
+  "points": 0,
+  "booksToOffer": [],
+  "booksToRemember": [],
+  "booksInterestedIn": [],
+  "matches": [],
+  "_id": "60a26618f8d1360733111027",
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2021-05-17T12:48:24.125Z",
+  "updatedAt": "2021-05-17T12:48:24.125Z"
+}
+``
+```
+
+### /:id
+
+**get** => response is an the user as an object
+**put** => If the field is made up =>
+
+```
+"This": "is not a field"
+```
```

Comment:
seems a bit confusing, this is about updating a user?
what you think about strukture like 
route
name of call 
req from frontend 
res from backend?

Reply:
added a new structure take a look and let me know :)